src/vegur_client.erl

```diff
@@ -690,7 +690,8 @@ stream_header(Client=#client{state=State, buffer=Buffer,
                             end
                     end;
                 <<"connection">> ->
-                    Values = header_list_values(Value),
```

Comment:
Yeah, can remove that function for sure.

It doesn't look like there's an existing test, adding one would be a lot of work to prove we're no longer hitting a bad arg in an edge case.